packages/flutter_tools/gradle/flutter.gradle

```diff
@@ -619,33 +620,70 @@ class FlutterPlugin implements Plugin<Project> {
             addApiDependencies(project, variant.name, project.files {
                 packFlutterAppAotTask
             })
-            // We know that the flutter app is a subproject in another Android app when these tasks exist.
             Task packageAssets = project.tasks.findByPath(":flutter:package${variant.name.capitalize()}Assets")
             Task cleanPackageAssets = project.tasks.findByPath(":flutter:cleanPackage${variant.name.capitalize()}Assets")
+            // In add to app scenarios, :flutter is a subproject of another Android app.
+            // We know that :flutter is used as a subproject when these tasks exist.
+            boolean isUsedAsSubproject = packageAssets && cleanPackageAssets
```

Comment:
Are there likely to be false positives or negatives from this? I'm not familiar with either of these, not sure how reliable this is.

Reply:
I don't think so since the `:flutter` project doesn't exist unless the app is using add 2 app and include_flutter.groovy.

I have also added an assert in line 654 to exit early if there isn't a root project named `:app`.